zipline/data/data_portal.py

```diff
@@ -643,6 +640,9 @@ def get_adjusted_value(self, asset, field, dt,
                                                  data_frequency)
 
         if isinstance(asset, Equity):
+            if data_frequency == 'daily':
+                dt = self.trading_calendar.session_open(dt)
```

Comment:
When in `daily` data frequency, I don't believe we should be introducing a dt which is not date normalized.

I think this bug shows that we should be adding a `data_frequency` parameter to `get_adjustments`, and making the query for `get_adjustments` aware of the `data_frequency`.

Reply:
I disagree with this recommendation.  `data_frequency` is a algorithm-level concept; we shouldn't be pushing that down to the layers of the stack that are purely about data retention and manipulation.  It's straightforward and context-independent to reason about what it means to say "give me data from time t0 adjusted as of time t1", whereas I have no intuition for what it means to say "give me data from time t0 at 'daily' data_frequency".

Reply:
@ssanderson you are right about keeping `DataPortal` free needing knowledge from the algorithm/backtest level data frequency. It had been a while since I've looked at this code, and had forgotten how important that is. Thanks for the admonition of keeping the levels separate.

However, I still believe that this is the wrong place to fix the application of adjustments when the dt and the perspective dt are on the same day.
The bug is that `get_adjustments` returns an adjustment when . I'll follow up with a suggestion how to fix that method.